CurrentPackage causes issues on startup

dtedder opened the pull request:
related to #418 

Shared/ToolResourceProvider.h
Shared/ToolResourceProvider.cpp
- created a new signal for `sceneDoneLoading`
- connected the scene's done loading to emit the new signal

Shared/messages/MessageFeedsController.h
Shared/messages/MessageFeedsController.cpp
- swapped the member variable for `GeoView` to `Scene`
- added slot for `setScene`
- removed the `setGeoView` method
- connect to `ToolResourceProvider::sceneDoneLoading` to `MessageFeedsController::setScene`
- guard any calls to `setupFeed` with a check to the `m_scene` pointer
- remove code no longer needed in `setupFeeds`for getting the scene

Shared/packages/OpenMobileScenePackageController.cpp
- set to const to fix clazy warning about detached container

dtedder committed:
Shared/ToolResourceProvider.h
Shared/ToolResourceProvider.cpp
- created a new signal for `sceneDoneLoading`
- connected the scene's done loading to emit the new signal

Shared/messages/MessageFeedsController.h
Shared/messages/MessageFeedsController.cpp
- swapped the member variable for `GeoView` to `Scene`
- added slot for `setScene`
- removed the `setGeoView` method
- connect to `ToolResourceProvider::sceneDoneLoading` to `MessageFeedsController::setScene`
- guard any calls to `setupFeed` with a check to the `m_scene` pointer
- remove code no longer needed in `setupFeeds`for getting the scene

Shared/packages/OpenMobileScenePackageController.cpp
- set to const to fix clazy warning about detached container

JamesMBallard commented on Shared/ToolResourceProvider.h:
```diff
@@ -81,6 +81,7 @@ public slots:
 signals:
   void sceneChanged();
   void geoViewChanged();
+  void sceneDoneLoading();
```
This does not seem like it should be a concern or a feature of the ToolResourceProvider.

JamesMBallard commented on Shared/messages/MessageFeedsController.cpp:
```diff
@@ -85,13 +82,22 @@ MessageFeedsController::~MessageFeedsController()
   \brief Sets the GeoView for the MessagesOverlay objects to \a geoView.
  */
-void MessageFeedsController::setGeoView(GeoView* geoView)
+void MessageFeedsController::setScene()
```
I would consider a different approach for this solution. It does appear there is an issue here if we start setting things up before scene is available, but this class is where that concern exists, so this class should be the one that can solve it.

There is a `sceneChanged` signal on the SceneQuickView you can listen for. Would that suffice here if there is no scene available yet?

dtedder replied:
SceneQuickView::sceneChanged is what I was looking for all along. Previously, I was attempting to get a reference to the `-arcGISScene()` in the `ToolResourceManager::sceneChanged` signal. This isn't the same of course so when a package was used to set the scene instead of the default scene it's ends up still null when the slot on MessageFeedController was called.

dtedder committed:
undoing change to ToolResourceProvider and using signal from SceneQuickView

Shared/ToolResourceProvider.h
Shared/ToolResourceProvider.cpp
- reverting the new signal and connection to the sceneDoneLoading

Shared/messages/MessageFeedController.h
Shared/messages/MessageFeedController.cpp
- switch back to the original connection to the GeoView event changed signal
- connecting to the ToolResourceProvider's GeoView's sceneChanged signal to assign the m_scene variable and call setupFeeds if it has not been called yet

JamesMBallard commented on Shared/messages/MessageFeedsController.h:
```diff
@@ -83,6 +81,9 @@ class MessageFeedsController : public AbstractTool
 
   static Esri::ArcGISRuntime::SurfacePlacement toSurfacePlacement(const QString& surfacePlacement);
 
+private slots:
+  void setScene();
```
A few things here. A bit pedantic, but we don't really need to specify "slots" anymore. They're just functions that we can connect directly to signals as you are doing. Optional.

Also I would consider a different name since it's unclear what is driving this call. Maybe `setSceneFromGeoview_();` or `onGeoviewChanged_();`

trailing underscore optional.

dtedder replied:
Pushed a new commit. removed the slots designation and just moved to be a private method. gave it a better name.

JamesMBallard reviewed (commented):
A few thoughts on naming, but otherwise looks good.

dtedder committed:
/Shared/messages/MessageFeedsController.cpp
/Shared/messages/MessageFeedsController.h
- removed slots identifier
- renamed to better indicate the actions of the method

dtedder commented:
redirecting this PR to the new issue created specifically for the crash on startup #450. Investigation into why the layers are not updating specifically on physical devices will continue in another branch.
